chapters/interface.tex

```diff
@@ -585,11 +585,20 @@ \section{Function-Compatibility or Function-Subtyping for Functions}\label{funct
 
 \section{Type Compatible Expressions}\label{type-compatible-expressions}
 
-Certain expressions consist of an operator applied to two or more type compatible subexpressions (\lstinline!A! and \lstinline!B!), including binary operators, e.g. \lstinline!A + B!, \lstinline!if!-expressions, e.g.\ \lstinline!if x then A else B!, and array expressions, e.g.\ \lstinline!{A, B}!.
+Certain expressions consist of an operator applied to two or more type compatible subexpressions (\lstinline!A! and \lstinline!B!).
+
+This includes:
+\begin{itemize}
+\item \lstinline!if!-expressions, e.g.\ \lstinline!if x then A else B!.
+\item Array expressions, e.g.\ \lstinline!{A, B}!
+\item Binary operators if both operands are of simple types, e.g.\ \lstinline!A + B!.
+Binary operators for other types are only defined for operator records, \cref{overloaded-binary-operations}, and do not necessarily require that the operands are type compatible with each other.
+\end{itemize}
 The resulting type of the expression in case of two type compatible subexpressions \lstinline!A! and \lstinline!B! is defined as follows:
```

Comment:
The list that follows also defines when A and B are considered type compatible, so this is not a very accurate statement. Realized when writing a comment for the line 624.

Reply:
Yes. I also noticed.

Reply:
I have now attempted to reformulate it. It is kind of awkward as we both use "type compatible" to say that `a` and `b` are type compatible in `{a,b}` and also for the type of `{a,b}`.

Reply:
Yes, I was going to comment on that too. I think there are places where we should say the "type of the resulting expression" instead of "type compatible expressions." Haven't looked at the latest changes yet.